packages/realm/src/Object.ts

```diff
@@ -120,13 +123,37 @@ class RealmObject<T = DefaultObject> {
   entries(): [string, unknown][] {
     throw new Error("Not yet implemented");
   }
-  toJSON(): unknown {
-    // return { ...this };
-    for (const key in this) {
-      const value = this[key];
-      console.log({ key, value });
+
+  /**
+   * Returns a plain object representation with possible circular references
+   * from the object for JSON serialization.
+   * @returns A plain object
+   */
+  toJSON(_?: string, cache = new JSONCacheMap<T>()): DefaultObject {
+    // Construct a reference-id of table-name & primaryKey if it exists, or fall back to objectId.
+
+    // Check if current objectId has already processed, to keep object references the same.
+    const existing = cache.find(this);
+    if (existing) {
+      return existing;
     }
-    return { ...this };
+    const result: DefaultObject = {};
+    cache.add(this, result);
+    // Move all enumerable keys to result, triggering any specific toJSON implementation in the process.
+    Object.entries(this).forEach(([key, value]) => {
```

Comment:
Reason why this is a `forEach` instead of a .`entries(...).map(...)` is because we need to add the object into cache before we start looping through it.